core/io/http_client.cpp

```diff
@@ -113,14 +128,14 @@ Error HTTPClient::request_raw(Method p_method, const String &p_url, const Vector
 	}
 	if (add_clen) {
 		request += "Content-Length: " + itos(p_body.size()) + "\r\n";
-		//should it add utf8 encoding? not sure
+		// Should it add utf8 encoding?
 	}
 	request += "\r\n";
 	CharString cs = request.utf8();
 
 	PoolVector<uint8_t> data;
 
-	//Maybe this goes faster somehow?
+	// Maybe this can be optimized?
```

Comment:
Since you are at it, I think you actually can make it faster with:
```
PoolVector<uint8_t> data;
data.resize(cs.length());
{
	PoolVector<uint8_t>::Write data_write = data.write();
	for (int i = 0; i < cs.length(); i++) {
			data_write[i] = cs[i];
	}
}
```
(mind the extra `{` `}` to create a block which then sends the `data_write` out of scope unlocking the memory)